packages/hash/integration/src/tests/model/knowledge/block.model.test.ts

```diff
@@ -0,0 +1,115 @@
+import { getRequiredEnv } from "@hashintel/hash-backend-utils/environment";
+import { createGraphClient } from "@hashintel/hash-api/src/graph";
+import { ensureWorkspaceTypesExist } from "@hashintel/hash-api/src/graph/workspace-types";
+import {
+  BlockModel,
+  EntityModel,
+  EntityTypeModel,
+  UserModel,
+} from "@hashintel/hash-api/src/model";
+import { generateWorkspaceEntityTypeSchema } from "@hashintel/hash-api/src/model/util";
+import { Logger } from "@hashintel/hash-backend-utils/logger";
+import { createTestUser } from "../../util";
+
+jest.setTimeout(60000);
+
+const logger = new Logger({
+  mode: "dev",
+  level: "debug",
+  serviceName: "integration-tests",
+});
+
+const graphApiHost = getRequiredEnv("HASH_GRAPH_API_HOST");
+const graphApiPort = parseInt(getRequiredEnv("HASH_GRAPH_API_PORT"), 10);
+
+const graphApi = createGraphClient(logger, {
+  host: graphApiHost,
+  port: graphApiPort,
+});
+
+describe("Block model class", () => {
+  let testUser: UserModel;
+
+  let testBlock: BlockModel;
+
+  const testBlockComponentId = "test-component-id";
+
+  let testBlockDataEntity: EntityModel;
+
+  let dummyEntityType: EntityTypeModel;
+
+  beforeAll(async () => {
+    await ensureWorkspaceTypesExist({ graphApi, logger });
+
+    testUser = await createTestUser(graphApi, "blockModelTest", logger);
+
+    dummyEntityType = await EntityTypeModel.create(graphApi, {
+      accountId: testUser.entityId,
+      schema: generateWorkspaceEntityTypeSchema({
+        namespace: testUser.getShortname()!,
+        title: "Dummy",
```

Comment:
I think Alfie wanted these tests to be somewhat meaningful in their definition (their names, variable names and such). Do you think we can find a good example test case that is easy to conceptually follow?

Reply:
I think in this instance it may actually make things more confusing, but I don't think we have a common understanding of what the entity type of the block data entity actually is (i.e. Is it the block schema? Can it be an arbitrary entity type like `Person`?).

`Dummy` keeps that ambiguous for now, I've added a `@todo` to highlight why `Dummy` was chosen and that it should be replaced in https://github.com/hashintel/hash/pull/1080/commits/a135b4cbab8c2400ee073064bde4d77b3dd31854